src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/constrainedtransfer/ConstrainedBoardingSearch.java

```diff
@@ -100,14 +100,19 @@ public boolean transferExist(int targetStopPos) {
         );
     }
 
-    private List<TransferForPattern> findMatchingTransfers(
+    private Iterable<TransferForPattern> findMatchingTransfers(
             TripSchedule tripSchedule,
             int stopIndex
     ) {
         final Trip trip = tripSchedule.getOriginalTripTimes().getTrip();
-        return currentTransfers.stream()
-                .filter(t -> t.matchesSourcePoint(stopIndex, trip))
-                .collect(Collectors.toList());
+        // for performance reasons we use a for loop here as streams are much slower
```

Comment:
```suggestion
        // for performance reasons we use a for loop here as streams are much slower.
        // I experimented with LinkedList and ArrayList and LinkedList was faster.
```
I think you should keep the comment about the LinkedList, because that is the "hidden" gold here. 
It is not obvious, that it is faster - it depends on the distribution of transfers and GC.